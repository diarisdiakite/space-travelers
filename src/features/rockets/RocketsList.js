import { React } from 'react';
import { useSelector } from 'react-redux';
import Rockets from './Rockets';
import { getAllRockets } from './rocketsSlice';

const RocketsList = () => {
  const { rockets } = useSelector(getAllRockets);
  return (
    <div>
      <ul>
        {rockets.map((rocket) => (
          <Rockets
            key={rocket.id}
            id={rocket.id}
            rocketName={rocket.name}
            description={rocket.description}
            flickrImages={rocket.flickr_images}
            reserved={rocket.reserved}
          />
        ))}
      </ul>
    </div>
  );
};

export default RocketsList;
