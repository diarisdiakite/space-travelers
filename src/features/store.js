import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
