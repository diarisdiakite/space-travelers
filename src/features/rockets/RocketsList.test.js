import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from '../store';
import RocketsList from './RocketsList';

describe('RocketsList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <RocketsList />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
