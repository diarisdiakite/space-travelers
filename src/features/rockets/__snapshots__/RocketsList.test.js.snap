// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsList component renders correctly 1`] = `
<div>
  <ul />
</div>
`;
